qiskit_dynamics/solvers/solver_classes.py

```diff
@@ -312,56 +312,39 @@ def solve(
             results.y = Array(results.y).reshape((len(results.y),) + y_input.shape, order="F")
 
         if y0_cls is not None:
-            results.y = [final_state_converter(yi, y0_cls) for yi in results.y]
+            results.y = [state_type_wrapper(yi) for yi in results.y]
 
         return results
 
 
-def initial_state_converter(
-    obj: Any, return_class: bool = False
-) -> Union[Array, Tuple[Array, Type]]:
-    """Convert initial state object to an Array.
+def initial_state_converter(obj: Any) -> Tuple[Array, Type, Callable]:
```

Comment:
@chriseclectic what are your thoughts on handling these parameters? Are they even meant to be mutable? I just tried creating an `Operator` and setting `atol` and it said it can't be set.

Reply:
The only one that matters is dims 

atol/rtol are class attributes, not instance attributes so don't matter in this case. They are set for all statevectors by doing `Statevector.atol = val`.